test/e2e/tests/test_monitoring_schedule.py

```diff
@@ -0,0 +1,219 @@
+        monitoring_schedule_name,
+    )
+
+
+def wait_resource_monitoring_schedule_status(
+    reference: k8s.CustomResourceReference,
+    expected_status: str,
+    wait_periods: int = 6,
+    period_length: int = 30,
+):
+    return wait_for_status(
+        expected_status,
+        wait_periods,
+        period_length,
+        get_monitoring_schedule_resource_status,
+        reference,
+    )
+
+
+@service_marker
+@pytest.mark.canary
+class TestMonitoringSchedule:
+    status_pending: str = "Pending"
+    status_scheduled: str = "Scheduled"
+
+    def _assert_monitoring_schedule_status_in_sync(
+        self,
+        sagemaker_client,
+        schedule_name,
+        reference,
+        expected_status,
+        wait_periods: int = 6,
+        period_length: int = 30,
+    ):
+        assert (
+            wait_sagemaker_monitoring_schedule_status(
+                sagemaker_client, schedule_name, expected_status
+            )
+            == wait_resource_monitoring_schedule_status(reference, expected_status, 2)
+            == expected_status
+        )
+
+    def test_smoke(self, sagemaker_client, xgboost_churn_data_quality_monitoring_schedule):
+        (reference, resource, spec) = xgboost_churn_data_quality_monitoring_schedule
+        assert k8s.get_resource_exists(reference)
+
+        monitoring_schedule_name = resource["spec"].get("monitoringScheduleName")
+
+        assert (
+            k8s.get_resource_arn(resource)
+            == describe_sagemaker_monitoring_schedule(
+                sagemaker_client, monitoring_schedule_name
+            )["MonitoringScheduleArn"]
+        )
+
+        # scheule transitions Pending -> Scheduled state
+        # Pending status is shortlived only for 30 seconds because baselining job has already been run
+        # remove the checks for Pending status if the test is flaky because of this
+        # as the main objective is to test for Scheduled status
+        # OR
+        # create the schedule with a on-going baseline job where it waits for the baselining job to complete
+        assert (
+            wait_resource_monitoring_schedule_status(
+                reference, self.status_pending, 5, 2
+            )
+            == self.status_pending
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False", 5, 2)
+
+        self._assert_monitoring_schedule_status_in_sync(
+            sagemaker_client, monitoring_schedule_name, reference, self.status_scheduled
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True")
+
+        # Update the resource
+        new_cron_expression = "cron(0 * * * ? *)"
+        spec["spec"]["monitoringScheduleConfig"]["scheduleConfig"][
+            "scheduleExpression"
+        ] = new_cron_expression
+        resource = k8s.patch_custom_resource(reference, spec)
+        resource = k8s.wait_resource_consumed_by_controller(reference)
+        assert resource is not None
```

Comment:
Considering the custom update logic, let's just put this into a separate test? That way you can have finer control over the tests for update rather than lumping it into smoke

Reply:
all resources now have some custom code which make it deviate from default generated code so I think this is ok and anyway these operations need to happen in order